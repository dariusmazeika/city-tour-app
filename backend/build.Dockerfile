ARG image
FROM $image

COPY . /srv/

WORKDIR /srv/
CMD ["sh", "./bin/startup.sh"]
