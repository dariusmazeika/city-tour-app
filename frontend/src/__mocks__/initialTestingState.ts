export default {
  localization: {
    currentLanguage: 'en',
  },
};
