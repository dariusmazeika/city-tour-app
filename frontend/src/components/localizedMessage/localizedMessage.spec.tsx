import { shallow } from 'enzyme/build';
import * as React from 'react';
import LocalizedMessage from './localizedMessage';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import initialTestingState from '../../__mocks__/initialTestingState';

describe('<LocalizedMessage />', () => {
  let store;
  beforeAll(() => {
    store = configureMockStore()(initialTestingState);
  });

  it('should match snapshot', () => {
    const wrapper = create(<Provider store={store}>
      <LocalizedMessage msg={'msg_key'}/>
    </Provider>);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('Shoud render key <LocalizedMessage />', () => {
  //   const key = 'msg_key';
  //   const wrapper = shallow(
  //     <Provider store={store}>
  //       <LocalizedMessage msg={key}/>
  //     </Provider>,
  //   );
  //
  //   const text = wrapper.find('span').text();
  //   expect(text).toEqual(key);
  //
  // });

  // it('Shoud find className if one is present', () => {
  //   const className = 'className';
  //   const wrapper = shallow(<Provider store={store}>
  //     <LocalizedMessage msg={''} className={className}/>
  //     </Provider>,
  //   );
  //   expect(wrapper.find('span').hasClass(className)).toEqual(true);
  //
  // });

});
