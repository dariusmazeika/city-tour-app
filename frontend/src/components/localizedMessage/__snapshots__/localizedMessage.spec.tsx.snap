// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocalizedMessage /> Shoud find className if one is present 1`] = `
<span
  className="this-is-css-class"
>
  msg_key
</span>
`;

exports[`<LocalizedMessage /> Shoud find className if one is present 2`] = `
<span
  className="this-is-css-class"
>
  translation
</span>
`;

exports[`<LocalizedMessage /> Should find a translation  1`] = `
<span
  className="this-is-css-class"
>
  translation
</span>
`;

exports[`<LocalizedMessage /> should match snapshot 1`] = `
<span>
  msg_key
</span>
`;
